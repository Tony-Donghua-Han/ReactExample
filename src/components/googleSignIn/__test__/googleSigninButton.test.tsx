import React from "react";
import { shallow } from "enzyme";
import GoogleSigninButton from "../googleSignInButton";

describe("Google Signin Button tests:", (): void => {
    test("Button should render without crushing", (): void => {
        const wrapper = shallow(<GoogleSigninButton onClick={(): void => {}} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test("Button should have 'Sign in with Google' displayed as text", (): void => {
        const wrapper = shallow(<GoogleSigninButton onClick={(): void => {}} />);
        expect(wrapper.find(".google-btn-text").text()).toEqual("Sign in with Google");
    });

    test("Clicking on the button would trigger the onClick event", (): void => {
        const mockOnClickHandler = jest.fn();
        const wrapper = shallow(<GoogleSigninButton onClick={mockOnClickHandler} />);
        wrapper.find(".google-signin").simulate("click");
        expect(mockOnClickHandler.mock.calls.length).toBe(1);
    });

    test("Press Enter key when button is focused should trigger the onClick event", (): void => {
        const mockOnClickHandler = jest.fn();
        const wrapper = shallow(<GoogleSigninButton onClick={mockOnClickHandler} />);
        wrapper.find(".google-signin").simulate("keydown", { keyCode: 13 });
        expect(mockOnClickHandler.mock.calls.length).toBe(1);
    });

    test("Press Space key (or any other key) when button is focused should not trigger the onClick event", (): void => {
        const mockOnClickHandler = jest.fn();
        const wrapper = shallow(<GoogleSigninButton onClick={mockOnClickHandler} />);
        wrapper.find(".google-signin").simulate("keydown", { keyCode: 32 });
        expect(mockOnClickHandler.mock.calls.length).toBe(0);
    });
});
