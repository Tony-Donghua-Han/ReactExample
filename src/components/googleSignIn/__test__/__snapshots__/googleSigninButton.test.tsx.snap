// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Signin Button Disabled should render without crushing 1`] = `
"<div className=\\"google-signin-disabled\\">
  <div className=\\"g-btn-icon\\" />
  <span className=\\"g-btn-text\\">
    Sign in with Google
  </span>
</div>"
`;

exports[`Google Signin Button Regular should render without crushing 1`] = `
"<div className=\\"google-signin\\" onClick={[Function: onClick]} onKeyDown={[Function: handleEnterKey]} role=\\"button\\" tabIndex={0}>
  <div className=\\"g-btn-icon\\" />
  <span className=\\"g-btn-text\\">
    Sign in with Google
  </span>
</div>"
`;
